core/src/main/java/hudson/FilePath.java

```diff
@@ -3338,7 +3339,13 @@ private int findSeparator(String pattern) {
     @Restricted(NoExternalUse.class)
     static class UrlFactory {
         public URL newURL(String location) throws MalformedURLException {
-            return new URL(location);
+            try {
```

Comment:
this is ok - but at the same time the class is not public so can only really be used inside the same package (especially since it is not extending anything public - except Object which does not not this method) (and the singleton instance is private). 
Can the callers be adjusted instead?
I only found one caller in a quick scan -> https://github.com/jenkinsci/jenkins/blob/778b8b62de4e6f42da87cd78dbe8d731a6888cf6/core/src/main/java/hudson/FilePath.java#L1013

Reply:
I am unable to understand your point here.

Reply:
> I am unable to understand your point here.

you could confidently let `URISyntaxException` be thrown from this method and adapt the callers as there can be none that are external to this package - and hence there should be done external to this repository.